File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        minPrice = prices[0]
        n = len(prices)
        for i in range(n):
            if prices[i] > minPrice:
                maxProfit = max(maxProfit,prices[i] - minPrice)
            else:
                minPrice = min(minPrice, prices[i])
        return maxProfit

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int maxProfit = 0;
        int n = prices.size();
        int minPrice = prices[0];

        for(int i = 0; i < n; i++){
            if(prices[i] > minPrice){
                maxProfit = max(maxProfit, prices[i] - minPrice);
            }
            minPrice = min(minPrice, prices[i]);
        }
        return maxProfit;
    }
};